### ISSUE TITLE
[prefer-optional-chain] Breaking change in v5.16.0 - requires type information

### ISSUE THREAD

**Issue by avaly** [typescript-eslint/typescript-eslint#4755]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/prefer-optional-chain": ["error"]
  }
}
```

```TS
// your repro code case
```

**Expected Result**

This rule used to work without type information before v5.16.0

**Actual Result**

Upgrading to v5.16.0, it now requires type information.

```
Error: Error while loading rule '@typescript-eslint/prefer-optional-chain': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
```

**Additional Info**

This is a breaking change and should be included in a major version bump.

Also, the documentation table on the rules do not mention this rule requires type information: https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/eslint-plugin

Regression introduced in #4430

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.16.0` |
| `@typescript-eslint/parser`        | `5.16.0` |
| `TypeScript`                       | `4.6.2` |
| `ESLint`                           | `8.11.0` |
| `node`                             | `14.19.1` |

**Comment by bradzacher** [comment]
The rule does not require type information. Our tests for the rule do not run with type information and are all passing.

Likely there is something else at play here.
Could you please provide an isolated reproduction for this?

**Comment by avaly** [comment]
Sorry for my initial assumption. Indeed it does not require type information. 

However when used together with the `graphql-eslint` plugin, it does trigger that error.

Reproducible with: https://gist.github.com/avaly/674f8bc818b2493ff54260090e9f73aa

```
$ eslint --ext graphql .

Oops! Something went wrong! :(

ESLint: 8.11.0

Error: Error while loading rule '@typescript-eslint/prefer-optional-chain': You have used a rule which requires parserServices to be generated. You must therefore provide a value for the "parserOptions.project" property for @typescript-eslint/parser.
Occurred while linting /tmp/eslint-bugs/index.graphql
    at Object.getParserServices (/tmp/eslint-bugs/node_modules/@typescript-eslint/utils/dist/eslint-utils/getParserServices.js:16:15)
    at create (/tmp/eslint-bugs/node_modules/@typescript-eslint/eslint-plugin/dist/rules/prefer-optional-chain.js:65:37)
    at Object.create (/tmp/eslint-bugs/node_modules/@typescript-eslint/utils/dist/eslint-utils/RuleCreator.js:42:20)
    at createRuleListeners (/tmp/eslint-bugs/node_modules/eslint/lib/linter/linter.js:922:21)
    at /tmp/eslint-bugs/node_modules/eslint/lib/linter/linter.js:1104:31
    at Array.forEach (<anonymous>)
    at runRules (/tmp/eslint-bugs/node_modules/eslint/lib/linter/linter.js:1041:34)
    at Linter._verifyWithoutProcessors (/tmp/eslint-bugs/node_modules/eslint/lib/linter/linter.js:1389:31)
    at Linter._verifyWithConfigArray (/tmp/eslint-bugs/node_modules/eslint/lib/linter/linter.js:1729:21)
    at Linter.verify (/tmp/eslint-bugs/node_modules/eslint/lib/linter/linter.js:1471:65)
```

If I disable the `prefer-optional-chain` rule, eslint runs without errors.

**Comment by rpastro** [comment]
I'm having the same error with version v5.16.0.

**Comment by bradzacher** [comment]
Sorry - to clarify.

Using our rules on files that are not TypeScript files or with parsers that are not `@typescript-eslint/parser` is considered "undefined" behaviour.

The fact that it used to work and now that it doesn't is not a breaking change because we never defined that our rules would work for this usecase. This is the danger of relying upon undefined behaviour - it may break at any time and without warning.

You can use [overrides](https://eslint.org/docs/user-guide/configuring/configuration-files#configuration-based-on-glob-patterns) to specify which rules should or should not be run on files.

**Comment by pleunv** [comment]
Fwiw, I ran into this issue as well, and took this as an opportunity to finally move my top-level plugins & rules to a JS/TS specific override so they do not by default also apply to other file types. It doesn't make a lot of sense to have @typescript-eslint and its rules run on GraphQL files, as @bradzacher says.

In essence, it means moving all your global config to a JS/TS override, i.e.

```
module.exports = {
  files: ['**/*.{js,ts,jsx,tsx}'],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    sourceType: 'module'
    project: './tsconfig.json'
  },
  plugins: ['@typescript-eslint'],
  rules: {
    // ...
  }
}
```
to
```
module.exports = {
  overrides: [
    {
      files: ['**/*.{js,ts,jsx,tsx}'],
      parser: '@typescript-eslint/parser',
      parserOptions: {
        sourceType: 'module'
        project: './tsconfig.json'
      },
      plugins: ['@typescript-eslint'],
      rules: {
        // ...
      }
    }
  ]
}
```

Everything should keep working as before.
